compiler/compile/OMRCompilation.cpp

```diff
@@ -2748,3 +2750,27 @@ bool OMR::Compilation::isRecursiveMethodTarget(TR::Symbol *targetSymbol)
 
    return isRecursive;
    }
+
+const TR::TypeLayout* OMR::Compilation::typeLayout(TR_OpaqueClassBlock * clazz)
+   {
+   TR::Region& region = self()->region();
+   auto it = _typeLayoutMap.find(clazz); 
+   if (it != _typeLayoutMap.end())
+      {
+      return it->second;
+      }
+   else
+      {
+      auto layout = TR::Compiler->cls.enumerateFields(region, clazz, self());
+      if (layout) 
+         {
+         _typeLayoutMap.insert(std::make_pair(clazz, layout)); 
+         return layout;
+         } 
+      else 
+         { 
+         _typeLayoutMap.insert(std::make_pair(clazz, (TR::TypeLayout*)NULL));
+         return NULL; 
+         }
```

Comment:
Is this if-else really needed? It seems to me that just having
```c++
auto layout = TR::Compiler->cls.enumerateFields(region, clazz, self());
_typeLayoutMap.insert(std::make_pair(clazz, layout)); 
return layout;
```
 would achieve the same thing.

Reply:
Right. I have corrected and updated this.